### PR TITLE
Check device api version before requesting extensions

### PR DESCRIPTION
**Pull request by nshoes**
Related: https://github.com/nerves-hub/nerves_hub_link/pull/252

Older versions of `nerves_hub_link` blow up when receiving an unknown message. This PR ensures that we only request extensions when we're communicating with a device that can handle the message.

Tested locally with `nerves_hub_link`.

### CODE REVIEW

File: lib/nerves_hub_web/channels/device_channel.ex [hunk 1]
```diff
@@ -49,8 +49,11 @@ defmodule NervesHubWeb.DeviceChannel do
 
     send(self(), :device_registation)
 
-    # Request device extension capabilities
-    push(socket, "extensions:get", %{})
+    # Request device extension capabilities if possible
+    # Earlier versions of nerves_hub_link don't have a fallback for unknown messages,
+    # so check version before requesting extensions
+    if is_safe_to_request_extensions?(params["device_api_version"]),
+      do: push(socket, "extensions:get", %{})
```
Comment (jjcarstens) [inline lib/nerves_hub_web/channels/device_channel.ex:56]:
```suggestion
    if is_safe_to_request_extensions?(socket.assigns.device_api_version),
      do: push(socket, "extensions:get", %{})
```
If we move this to after all the assigns, we can piggy back on the safe default that `assign_api_version` gives us

File: lib/nerves_hub_web/channels/device_channel.ex [hunk 2]
```diff
@@ -612,4 +615,6 @@ defmodule NervesHubWeb.DeviceChannel do
       0
     end
   end
+
+  defp is_safe_to_request_extensions?(version), do: Version.match?(version, ">= 2.5.2")
```
Comment (jjcarstens) [inline lib/nerves_hub_web/channels/device_channel.ex:619]:
```suggestion
  defp is_safe_to_request_extensions?(version), do: Version.match?(version, "> 2.5.2")
```
`2.5.2` is the currently released NHL version, but the extensions work has not been released to hex. So we probably want that as non-inclusive greater-than to capture the next version which should have the extensions work

File: lib/nerves_hub_web/channels/device_channel.ex (hunk 1 above)
Comment (nshoes) [reply lib/nerves_hub_web/channels/device_channel.ex:56]: Gosh dangit lol Thanks for the callout.

**Review by jjcarstens — APPROVED** [review]

**Comment by jjcarstens** [comment]
Whoops. Missed the failing tests. Assuming you get those worked out, then this is 👍🏼

**Pull request merged**